diff --git a/flk b/flk
index c7455df..33ff99c 100755
--- a/flk
+++ b/flk
@@ -1794,6 +1794,22 @@ _file_exists () {
 
 _fref "file-exists?" _file_exists
 
+_file_write () {
+    local a="${__false}"
+    local f="${ANON["${1}"]}"; shift
+    local c="${ANON["${1}"]}"; shift
+    [ "${1}" != "" ] && a="${1}"; shift
+
+    if [ "${a}" == "${__true}" ]; then
+        echo -n "${c}" >> "${f}"
+    else
+        echo -n "${c}" > "${f}"
+    fi
+    r="${__nil}"
+}
+
+_fref "file-write" _file_write
+
 REP "(def! load-file-without-hashbang (fn* (f) (eval (read-string (str \"(do \" (remove-hashbang (slurp f) ) \"\nnil)\")))))"
 read -d "" __FLECK__REPCAPTURE << __FLECK__INLINEMALFILE
 ;; aliases for common clojure names to mal builtins
