--- flk	2020-12-21 21:01:18.452985058 -0300
+++ flk_new	2020-12-21 21:00:23.769273394 -0300
@@ -1794,6 +1794,18 @@
 
 _fref "file-exists?" _file_exists
 
+_dir_exists () {
+    local f="${ANON["${1}"]}"; shift
+
+    if [ -d "$f" ]; then
+        r="${__true}"
+    else
+        r="${__false}"
+    fi
+}
+
+_fref "dir-exists?" _dir_exists
+
 _file_write () {
     local a="${__false}"
     local f="${ANON["${1}"]}"; shift
@@ -1818,6 +1830,54 @@
 
 _fref "unlink" _unlink
 
+_list_dir () {
+    local p="./"
+    [ "${1}" != "" ] &&  p="${ANON["${1}"]}"; shift
+
+    if [ "${p: -1}" != "/" ]; then
+        p="${p}/"
+    fi
+
+    if [ ! -d "${p}" ]; then
+        r="${__false}"
+    else
+        _list
+        local newlist="${r}"
+
+        for i in ${p}*; do
+            if [ "${i}" != "${p}*" ]; then
+                local f=true
+
+                _hash_map
+                local newmap="${r}"
+                _string "$i"
+                _assoc! "${newmap}" "${__keyw}object" "${r}"
+
+                if [ -d "${i}" ]; then
+                    _string "directory"
+                else
+                    local fname="${i##*/}"
+                    local ext="${fname##*.}"
+                    if [ "$ext" == "${fname}" ]; then
+                        _string "file"
+                    else
+                        _string "$ext"
+                    fi
+                fi
+
+                _assoc! "${newmap}" "${__keyw}type" "${r}"
+                r="${newmap}"
+            else
+                r="${__nil}"
+            fi
+            _conj! "${newlist}" "${r}"
+        done
+        r="${newlist}"
+    fi
+}
+
+_fref "list-dir" _list_dir
+
 REP "(def! load-file-without-hashbang (fn* (f) (eval (read-string (str \"(do \" (remove-hashbang (slurp f) ) \"\nnil)\")))))"
 read -d "" __FLECK__REPCAPTURE << __FLECK__INLINEMALFILE
 ;; aliases for common clojure names to mal builtins
